geowrangler/datasets/nightlights.py

```diff
@@ -0,0 +1,526 @@
+            logger.info(f"Creating access token directory {save_path.parent}")
+            save_path.parent.mkdir(mode=510, parents=True, exist_ok=True)
+        with open(save_path, "w") as f:
+            f.write(access_token)
+    if set_env:
+        logger.info(f"Adding access token to environmentt var {env_token_var}")
+        os.environ[env_token_var] = access_token
+
+    return access_token
+
+
+# Cell
+
+
+def clear_eog_access_token(
+    save_file=DEFAULT_EOG_CREDS_PATH,
+    env_var=EOG_ENV_VAR,
+    clear_file=True,
+    clear_env=True,
+):
+    save_path = Path(os.path.expanduser(save_file))
+
+    if clear_file and save_path.exists():
+        logger.info(f"Clearing eog access token file {save_file}")
+        save_path.unlink()
+    if clear_env:
+        logger.info(f"Clearing eog access token environment var {env_var}")
+        os.environ[env_var] = ""
+
+
+# Cell
+
+
+def setup_eog_auth_headers(headers, access_token, env_var, creds_file):
+    if access_token is None:
+        # try getting it from environ
+        if (
+            os.environ.get(env_var, None) is not None
+            and len(os.environ.get(env_var)) > 0
+        ):
+            logger.info(f"Using access token from environment var {env_var}")
+            access_token = os.environ.get(env_var)
+        else:
+            save_path = Path(os.path.expanduser(creds_file))
+            if save_path.exists():
+                logger.info(f"Using access token from saved file {save_path}")
+                with open(save_path) as f:
+                    access_token = f.read()
+
+    if access_token:
+        auth = "Bearer " + access_token
+        if headers:
+            headers.update(dict(Authorization=auth))
+        else:
+            headers = dict(Authorization=auth)
+    return headers
+
+
+# Cell
+
+
+def download_url(
+    url,
+    dest=None,
+    access_token=None,
+    headers=None,
+    timeout=None,
+    show_progress=True,
+    chunksize=1024 * 1024,
+    env_var=EOG_ENV_VAR,
+    creds_file=DEFAULT_EOG_CREDS_PATH,
+):
+    "Download `url` to `dest` and show progress"
+    reporthook = make_report_hook(show_progress)
+
+    headers = setup_eog_auth_headers(headers, access_token, env_var, creds_file)
+
+    dest = urldest(url, dest)
+    if not dest.parent.is_dir():  # parent dir should always exist
+        dest.parent.mkdir(parents=True, exist_ok=True)
+
+    nm, resp, fp = urlretrieve(
+        url,
+        filename=dest,
+        headers=headers,
+        reporthook=reporthook,
+        timeout=timeout,
+        chunksize=chunksize,
+    )
+    if "Cache-Control" in resp and "must-revalidate" in resp["Cache-Control"]:
+        raise HTTPError(
+            url,
+            401,
+            "No access token or invalid access token provided, please call `get_eog_access_token` to get one",
+            resp,
+            fp,
+        )
+    return nm
+
+
+# Cell
+
+
+def unzip_eog_gzip(gz_file, dest=None, delete_src=False):
```

Comment:
not super sure if we need this works (based on docs)
```
with gzip.open('...gz') as gzip_infile:
    with rasterio.open(gzip_infile) as raster_infile:
        raster ...
```

Reply:
Yeah, but its a lot slower. Unlike the shapefile access where the degradation was minimal, using the rasterstats module (which wasn't reading the whole tif file into memory)  to generate zonal stats on the gzipped tif file was slower. Also, the unzipped tif file wasn't that much larger than the gzipped tif file.  (eg 11G vs 9.1G)

Reply:
alright got it. i wonder why it was gzipped if the files are around the same